### DISCUSSION TITLE
how to use mitmproxy's config.yaml

### DISCUSSION THREAD

**Discussion by SnowAugust** [mitmproxy/mitmproxy#6284] [Q&A]
I want to use Mitmproxy to build a proxy server to meet the requirements of connecting a corporate LAN to the Internet.
I need to meet the following requirements：

1. Only allowed devices in the enterprise LAN can connect to proxy servers.
2. Only specific URLs can be accessed, all others are blocked.  (e.g. _https://github.com/mitmproxy/*_ can be accessed, but _https://github.com/*_ need blocked)
3. All request and response contents need store to log files.
4. I hope to use a configuration file (config.yaml) to achieve the above requirements.

The challenges I am facing now:
1. Unable to find template for config.yaml
2. I don't know which configurations meet the above requirements

Please help me~~

**Comment by ThomasBucaioni** [comment]
Hey,
It rather looks like you need a regular proxy like nginx: https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/
The Mitmproxy is to "spy" Internet frames, but not to filter corporate traffic.
There's a nice course about HA proxy on [ACG](https://www.pluralsight.com/cloud-guru/courses/hands-on-with-haproxy-load-balancer), but maybe it's an overkill for what you need
